app/controllers/api/predictions_controller.rb

```diff
@@ -36,16 +36,18 @@ def update
     private
 
     def authorize_to_see_prediction
-      raise UnauthorizedRequest unless @prediction.public? || @user.authorized_for(@prediction)
+      raise UnauthorizedRequest unless @prediction.visible_to_everyone? ||
+                                       @user.authorized_for(@groups, @prediction)
     end
 
     def authorize_to_update_prediction
-      raise UnauthorizedRequest unless @user.authorized_for(@prediction)
+      raise UnauthorizedRequest unless @user.authorized_for(@groups, @prediction)
     end
 
     def build_new_prediction
-      unless prediction_params[:private] && @user
-        prediction_params[:private] = @user.private_default
+      permitted_params = prediction_params
+      if permitted_params[:visibility].nil? && @user.present?
+        permitted_params[:visibility] = @user.visibility_default
```

Comment:
I think:
`prediction_params[:visibility] ||= @user.try(:visibility_default)`
is more defensive